### PR TITLE
Add windows-2012 support

### PR DESCRIPTION
**Pull request by mikemadden42**

### CODE REVIEW

**Comment by mikemadden42** [comment]
jenkins run the tests please

**Comment by mikemadden42** [comment]
jenkins run the tests please

File: Jenkinsfile [hunk 1]
```diff
@@ -92,7 +92,7 @@ pipeline {
         }
 
         stage('Elastic Agent x-pack Windows'){
-          agent { label 'windows-immutable && windows-2019' }
+          agent { label 'windows-immutable && windows-2012' }
```
Comment (v1v) [inline Jenkinsfile:95]:
```suggestion
          agent { label 'windows-immutable && windows-2012-r2' }
```

And in the other sections

**Comment by mikemadden42** [comment]
jenkins run the tests please

**Comment by mikemadden42** [comment]
Closing the PR to focus on a better approach:
https://github.com/elastic/beats/pull/18458

**Pull request closed without merging**